src/js/constants.js

```diff
@@ -152,11 +152,24 @@ export const PointerTypes = {
   TOUCH: 'touch',
 };
 
+/**
+ * @type {{
+ *   UNAVAILABLE: 'unavailable';
+ *   UNSUPPORTED: 'unsupported';
+ * }}
+ */
```

Comment:
TS didn't derive this?
that's a shame

Reply:
Yeah, it's a bit more "loosey goosey" between the .js and the default inferred object typings. If this were actual typescript, the solution is pretty terse:
```ts
export const AvailabilityStates = {
  UNAVAILABLE: 'unavailable',
  UNSUPPORTED: 'unsupported',
} as const; // the as const does all the work of e.g. inferring keys as literal types.
```

Reply:
Yeah between this being .js and the default behavior for inferred types on complex objects, it ends up being a bit more "loosey goosey". If this were written in TypeScript, the solution would be much more terse/simple 💔, e.g.:
```ts
const AvailabilityStates = {
    UNAVAILABLE: 'unavailable',
    UNSUPPORTED: 'unsupported',
} as const; // the `as const` does all the work of e.g. inferring keys as literal types.
```